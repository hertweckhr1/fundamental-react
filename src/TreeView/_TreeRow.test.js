import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import TreeRow from './_TreeRow';
import TreeView from './TreeView';

// mock shortid for snapshot testing
jest.mock('shortid', () => {
    let id = 1;

    return {
        generate: () => id++
    };
});

describe('<TreeRow />', () => {
    const treeRow = (
        <TreeRow>
            <TreeView.Col>First Level</TreeView.Col>
            <TreeView.Col>Data Col 2</TreeView.Col>
            <TreeView.Col>Data Col 3</TreeView.Col>
            <TreeView.Col>Data Col 4</TreeView.Col>
        </TreeRow>
    );

    test('create tree row component', () => {
        let component = renderer.create(treeRow);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the TreeRow component', () => {
            const element = mount(
                <TreeRow
                    data-sample='Sample' />);

            expect(
                element.find('.fd-tree__row').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
