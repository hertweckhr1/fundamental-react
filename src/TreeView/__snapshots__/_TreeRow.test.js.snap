// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreeRow /> create tree row component 1`] = `
<div
  className="fd-tree__row"
>
  <div
    className="fd-tree__col fd-tree__col--control"
  >
    First Level
  </div>
  <div
    className="fd-tree__col"
  >
    Data Col 2
  </div>
  <div
    className="fd-tree__col"
  >
    Data Col 3
  </div>
  <div
    className="fd-tree__col"
  >
    Data Col 4
  </div>
</div>
`;
